### ISSUE TITLE
issue with distillation

### ISSUE THREAD

**Issue by huydang2106** [VT-NLP/Incremental_Prompting#1]
I have run the code and found that when doing distillation, data instances are forwarded through old model (with parameter stored in model.history) but the embedding layer params weren't stored, so the old model is actually called with new prompt embedding.
Is this a mistake or your purpose?

**Comment by mqianliu** [comment]
Thanks for raising this question. In our approach we did not apply distillation on new data ([Line 313-350 in `emp.py`](https://github.com/VT-NLP/Incremental_Prompting/blob/main/models/emp.py#L313)), instead we only used exemplar distillation (Line 351-381). In exemplar distillation, we directly stored the feature and logits in the previous task (`self.exemplar_logit` and `self.exemplar_feature`) when we selected examples for replaying so that there won't be inconsistency between the prompt and the rest of the old model.

It seems we set the `distill` argument to `True` in the uploaded version ([Line 131 in `run.py`](https://github.com/VT-NLP/Incremental_Prompting/blob/main/run.py#L131)). This should be set to `False` (fixed in the latest commit).

**Comment by huydang2106** [comment]
Thanks for the answer. I have another question. When call self.replay_forward, the function just forward the span representation to mlp, and then the final linear and prompt embedding is not forwarded here. Why it is so different with forward call with new data?

**Comment by mqianliu** [comment]
Thanks for asking! The difference between the forward of new data and replay is whether we apply an additional disentangle layer (`self.prompt_map`) and combine the logits to better tune the prompt embedding (it's also tuned even without the disentanglement). During a new task, we froze learned prompts' parameters to defy forgetting, and thus, we don't need the disentanglement to tune the learned prompts. We empirically found that the model performs slightly better with such a design. Also note that the prompt embedding is always attached to the input in BERT's embedding layer no matter the model is being trained on new data or replaying.

Hope this helps.

**Comment by mqianliu** [comment]
It seems there's no further question so I'll close this issue.
